instrumentation/iceberg-1.8/testing/src/main/java/io/opentelemetry/instrumentation/iceberg/v1_8/AbstractIcebergTest.java

```diff
@@ -0,0 +1,374 @@
+                                        .hasCount(1)
+                                        .hasAttributesSatisfying(
+                                            attributes ->
+                                                assertEquals(
+                                                    Attributes.builder()
+                                                        .put(
+                                                            "iceberg.schema.id",
+                                                            expectedReport.schemaId())
+                                                        .put(
+                                                            "iceberg.table.name",
+                                                            expectedReport.tableName())
+                                                        .put(
+                                                            "iceberg.snapshot.id",
+                                                            expectedReport.snapshotId())
+                                                        .build(),
+                                                    attributes)))));
+  }
+
+  private void assertIcebergCounterMetric(
+      String otelMetricName, String expectedUnit, ScanReport expectedReport, long expectedValue) {
+    testing()
+        .waitAndAssertMetrics(
+            "io.opentelemetry.iceberg-1.8",
+            metricAssert ->
+                metricAssert
+                    .hasName(otelMetricName)
+                    .hasUnit(expectedUnit)
+                    .hasLongSumSatisfying(
+                        sum ->
+                            sum.hasPointsSatisfying(
+                                longSumAssert ->
+                                    longSumAssert
+                                        .hasValue(expectedValue)
+                                        .hasAttributesSatisfying(
+                                            attributes ->
+                                                assertEquals(
+                                                    Attributes.builder()
+                                                        .put(
+                                                            "iceberg.schema.id",
+                                                            expectedReport.schemaId())
+                                                        .put(
+                                                            "iceberg.table.name",
+                                                            expectedReport.tableName())
+                                                        .put(
+                                                            "iceberg.snapshot.id",
+                                                            expectedReport.snapshotId())
+                                                        .build(),
+                                                    attributes)))));
```

Comment:
when possible we prefer to use satisfiesExactly

```suggestion
                                        .hasAttributesSatisfyingExactly(
                                            equalTo(longKey("iceberg.schema.id"), expectedReport.schemaId()),
                                            equalTo(stringKey("iceberg.table.name"), expectedReport.tableName()),
                                            equalTo(longKey("iceberg.snapshot.id"), expectedReport.snapshotId())
                                        ))));
```